bpf/include/bpf/ctx/xdp.h

```diff
@@ -271,6 +271,8 @@ ctx_adjust_hroom(struct xdp_md *ctx, const __s32 len_diff, const __u32 mode,
 			else
 				ret = -EFAULT;
 			break;
+		case -8:  /* __u32 opt[2] */
+			break;
```

Comment:
Note that this is a special-purpose implementation of `ctx_adjust_hroom()`, and doesn't behave as you might expect.

1. your code would currently strip off the first 8 bytes of the L2 header. I believe that's not what you want here.
2. there's an earlier `build_bug_on(len_diff <= 0)` check, so this won't even compile today.